pulser/sequence.py

```diff
@@ -321,7 +324,7 @@ def declare_channel(self, name: str, channel_id: str,
                 initial_target = None
             else:
                 # "_target" call is not saved
-                self._target(cast(Union[Iterable,
+                self._target(cast(Union[Set[QubitId],
```

Comment:
Don't forget this casting doesn't work before Python 3.9... Just leave `Iterable` as before, it's fine as an argument to `_target`

Reply:
`pulser\sequence.py:327: error: Argument 1 to "_target" of "Sequence" has incompatible type "Union[Iterable[Union[int, str]], Union[int, str]]"; expected "Union[AbstractSet[Union[int, str]], Union[int, str]]"`

Reply:
Yes, you have to change back to `Iterable` on the `_target` type annotation too